docs/adding_packages/build_and_package.md

```diff
@@ -13,24 +13,59 @@ Both methods often use build helpers to build binaries and install them into the
+  * `tools.build:cflags`
+  * `tools.build:cxxflags`
+  * `tools.build:defines`
+  * `tools.build:sharedlinklags`
+  * `tools.build:exelinkflags`
+  * `tools.apple:enable_bitcode` (only if host OS is `iOS`/`watchOS`/`tvOS`)
+
+* If host OS is Apple OS family (`macOS`/`iOS`/`watchOS`/`tvOS`), `-headerpad_max_install_names` flag should be passed to linker, unless host OS is `iOS`/`watchOS`/`tvOS` and `tools.apple:enable_bitcode` is enabled.
+
+* Multithread build (if supported by underlying build system):
+  * if some steps are sensitive to race conditions, monothread should be enforced.
+  * otherwise multithreaded build should be enabled with a number of cores controlled by `tools.build:jobs` conf from host profile if it is set, otherwise to all cores of build machine.
 
 ## Package
 
-* CMake config files must be removed (they will be generated for consumers by `cmake_find_package`, `cmake_find_package_multi`, or `CMakeDeps` generators). Use `rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))` or `rm(self, "*.cmake", os.path.join(self.package_folder, "lib"))`.
+* CMake config files must be removed. They will be generated for consumers by `CMakeDeps` generator (or legacy `cmake_find_package`/`cmake_find_package_multi` generators).
 
-* pkg-config files must be removed (they will be generated for consumers by `pkg_config` or `PkgConfigDeps` generators). Use `rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))` or `rm(self, "*.pc", os.path.join(self.package_folder, "lib"))`.
+* pkg-config files must be removed. They will be generated for consumers by `PkgConfigDeps` generator (or legacy `pkg_config` generator).
 
-* On *nix systems, executables and shared libraries should have empty `RPATH`/`RUNPATH`/`LC_RPATH`.
+* On *nix systems, executables and shared libraries should have empty `RPATH`/`RUNPATH`/`LC_RPATH`. Though, a relative path pointing inside package itself is allowed.
 
-* On macOS, install name in `LC_ID_DYLIB` section of shared libs must be `@rpath/<libfilename>`.
+* On Apple OS family:
+  * shared libs: name field of `LC_ID_DYLIB` load command must be `@rpath/<libfilename>`.
+  * shared libs & executables: name field of each `LC_LOAD_DYLIB` load command should be `@rpath/<libdependencyfilename>` (except those refering to system libs or frameworks).
 
 * Installed files must not contain absolute paths specific to build machine. Relative paths to other packages is also forbidden since relative paths of dependencies during build may not be the same for consumers. Hardcoded relative paths pointing to a location in the package itself are allowed.
+
+* Static and shared flavors of the same library must not be packaged together.
+
+* Usual package layout should be:
+
+  ```
+  bin/
+    - dll (Windows only)
+    - executables
+  include/
+    - headers
+  lib/
+    - static libs
+    - shared libs (not Windows)
+    - import libs of dll (Windows only)
+  res/
```

Comment:
I appreciate this is reflecting an already existing convention within Conan Center, but is there a definition of what constitutes a resource file here?  The rest of the layout seems to roughly follow the convention laid out here: https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.pdf , makes me wonder why `res` and not `share`, if indeed the purpose is for the same files.

Reply:
Resource file can be anything: a file required at runtime that we must reference in env var, some configurations files which can be used for build of other libraries, source code etc
For the reason why it's res instead of share, it's a question for conan authors, they have enforced this convention in conan and it has leaked in conan-center.

Reply:
`res` refers in general to the architecture-independent data, so it may contain anything, besides binaries/machine code. it may include (but not limited to):
- documentation, man pages
- graphics, like icons, cursors, images, animations
- sounds and music
- video
- databases
- configuration files (xml, json, etc.)
- translations

Reply:
@jcar87 I can remove this paragraph to not block PR. It's premature to write something regarding this topic if https://github.com/conan-io/conan/issues/12694 is not clarified.